app/assets/javascripts/discourse/controllers/login.js.es6

```diff
@@ -23,14 +24,23 @@ export default Ember.Controller.extend(ModalFunctionality, {
   authenticate: null,
   loggingIn: false,
   loggedIn: false,
+  secondFactorRequired: false,
 
   canLoginLocal: setting('enable_local_logins'),
   loginRequired: Em.computed.alias('application.loginRequired'),
 
+  @observes('secondFactorRequired')
```

Comment:
We try to avoid observers wherever possible.

In this case, you know exactly when you are updating `secondFactorRequired` - you could just call this function when you do! Or even have a function `updateSecondFactorRequired(required)` that would set it too.